plugins/modules/mysql_user.py

```diff
@@ -475,7 +477,19 @@ def user_add(cursor, user, host, host_all, password, encrypted,
     mogrify = do_not_mogrify_requires if old_user_mgmt else mogrify_requires
 
     if password and encrypted:
-        cursor.execute(*mogrify("CREATE USER %s@%s IDENTIFIED BY PASSWORD %s", (user, host, password), tls_requires))
+        server_version = get_server_version(cursor)
+
+        # The IDENTIFIED BY PASSWORD syntax was dropped in MySQL 8.0, but appears to be supported by MariaDB 10.x.
+        if server_version < LooseVersion("8") or server_version > LooseVersion("10"):
```

Comment:
@steveteahan hi, sorry for the delayed review - very long public holidays in my country.
Thanks for fixing this!

It might break down when MySQL becomes 10
```suggestion
        if server_version < LooseVersion("8") or server_version > LooseVersion("10"):
```
We can determine the condition above more precisely. As an example, look at `uses_deprecated_terminology` function [here](https://github.com/ansible-collections/community.mysql/pull/71/files)

What do you think?
(I'll try to respond much faster since the holidays have ended)

Reply:
That makes sense to me.  I'll leave the `get_server_version` because I think that could be used to get rid of some duplicate logic (I'll keep that for another review).